types/react-navigation/index.d.ts

```diff
@@ -1161,8 +1161,8 @@ export interface NavigationInjectedProps {
 }
 
 export function withNavigation<T = {}>(
-  Component: React.ComponentType<T & NavigationInjectedProps>
-): React.ComponentType<T>;
+  Component: React.ComponentClass<T & NavigationInjectedProps>
```

Comment:
Why change to `ComponentClass` here? Is this HOC only for class-based component?

Reply:
Oops my bad, not intentional.

Reply:
it's been fixed